code/controllers/subsystems/cleanup_ch.dm

```diff
@@ -0,0 +1,229 @@
+#define DISABLE_CLEANUP (1<<12)
+#define FIRE_PRIORITY_CLEANUP 120
+#define MAX_CLEANUP_QUEUE_TIME 120 //Max time in queue before we give up.
+#define URGENCY_PANIC_MODE 0.75 //Max urgency before we start to skip checks
+#define SMOOTHSTEP(variable) (max(min(6*(variable**5) - 15*(variable**4) + 10*(variable**3),1),0))
+//This is the cleanup subsystem. Its responsibility is to clean up unneeded atoms such as empty bullet casings and dead mobs in an effort to boost performance.
+SUBSYSTEM_DEF(cleanup)
+	name = "Cleanup"
+	priority = FIRE_PRIORITY_CLEANUP
+	flags = SS_NO_INIT
+	wait = 2 MINUTES //Fires every 2 minutes.
+
+	var/stat_tag = "Cl" //Used for logging
+	var/urgency //How desperate are we to delete stuff?
+	var/points //How much complexity is in our queue?
+	
+
+	var/list/processing = list()
+	var/list/currentrun = list()
+	
+	var/list/stinkyqueries = list() //I hate this, but for the attack logs SQL queries, we need to check them otherwise the threads won't close.
+	var/list/currentqueries = list()
+
+	var/datum/cleanupentry/current_thing
+	var/current_query
+
+/datum/controller/subsystem/cleanup/Recover()
+	log_debug("[name] subsystem Recover().")
+	if(SScleanup.current_thing)
+		log_debug("current_thing was: (\ref[SScleanup.current_thing])[SScleanup.current_thing]([SScleanup.current_thing.type]) - currentrun: [SScleanup.currentrun.len] vs total: [SScleanup.processing.len]")
+	var/list/old_processing = SScleanup.processing.Copy()
+	for(var/datum/D in old_processing)
+		if(istype(D))
+			processing |= D
+
+/datum/controller/subsystem/cleanup/stat_entry()
+	..("[stat_tag]:[processing.len] U:[urgency] P:[points]")
+
+/datum/controller/subsystem/cleanup/proc/get_points()
+	var/temp_points = processing.len
+	for(var/mob/mob_entry in processing)
+		temp_points += 2 //Mobs are worth 3x as much as atoms, since they're counted once in the first definition for temp_points, and twice here.
+	for(var/mob/living/carbon/carbon_entry in processing)
+		temp_points += 2 //Carbons are worth 5x as much as atoms, since they're counted once in the first definition for temp_points, twice for being a mob, and twice for carbon.
+	return temp_points
+
+/proc/cleanup_adjusted_points(points)
+	return (log(points)-2.5)/3
```

Comment:
This spot runtimes when points is 0

Reply:
oopsie poopsie, I really should have predicted this -w-

Cheers.